File: EAGetMail/Samples_NetDesktop/VB/ImapFull/AppProject1/ListMailItemCompare.vb
Class ListMailItemCompare
    Implements IComparer
    ' Compare for ListViewMail sort
    Public Function IComparer_Compare(x As Object, y As Object) As Integer Implements IComparer.Compare
        Dim itemx As ListViewItem = TryCast(x, ListViewItem)
        Dim itemy As ListViewItem = TryCast(y, ListViewItem)

        Dim sx As String = itemx.SubItems(2).Text
        Dim sy As String = itemy.SubItems(2).Text

        If sx.Length <> sy.Length Then Return -1

        Dim count As Integer = sx.Length
        For i As Integer = 0 To count - 1

            If sx(i) > sy(i) Then
                Return -1
            ElseIf sx(i) < sy(i) Then
                Return 1
            End If
        Next

        Return 0
    End Function
End Class

File: EAGetMail/Samples_NetDesktop/CSharp/Simple/AppProject1/ListMailItemCompare.cs
using System;
using System.Collections;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace AppProject1
{
    class ListMailItemCompare: IComparer
    {
        // Sort the email as received data.
        public int Compare(object x, object y)
        {
            ListViewItem itemx = x as ListViewItem;
            ListViewItem itemy = y as ListViewItem;

            string sx = itemx.SubItems[2].Text;
            string sy = itemy.SubItems[2].Text;
            if (sx.Length != sy.Length)
                return -1; //should never occured.

            int count = sx.Length;
            for (int i = 0; i < count; i++)
            {
                if (sx[i] > sy[i])
                    return -1;
                else if (sx[i] < sy[i])
                    return 1;
            }

            return 0;
        }
    }
}
